definitions/features/capsule.rb

```diff
@@ -0,0 +1,31 @@
+class Features::Capsule < ForemanMaintain::Feature
+  include ForemanMaintain::Concerns::Downstream
+  include ForemanMaintain::Concerns::BaseForemanProxy
+
+  metadata do
+    label :capsule
+
+    confine do
+      find_package('foreman-proxy') && feature(:instance).downstream?
+    end
+  end
+
+  def internal?
+    !!feature(:foreman_server)
+  end
+
+  def external?
```

Comment:
Is there anything else besides internal and external? Why external? is not !internal? Do we need it at all?

Reply:
Inside `internal?` method, I wanted to avoid checking installer's last_scenario. As we have most of the checks common to internal as well as external capsule where having a feature capsule or downstream? will be enough.
But upgrade scenarios will be valid in case of `feature(:capsule).external?` where it's worth checking installer's last_scenario. 
Please correct me if I am missing anything here.

Reply:
I'm not that familiar with what we need to distinguish so let me sum up my expectation and then work together on clarifying it:

| Case             | feature                   | detection                                        |                                  |
|------------------|---------------------------|--------------------------------------------------|----------------------------------|
| Proxy            | :foreman_proxy            | package 'foreman-proxy'                          | Includes also downstream proxies |
| Internal Proxy   | :foreman_proxy.internal?  | package 'foreman-proxy' && package 'foreman'     |                                  |
| External Proxy   | !:foreman_proxy.internal? |                                                  |                                  |
| Capsule          | :capsule                  | package 'satellite-capsule'                      | Downstream only                  |
| Internal Capsule | :capsule.internal?        | package 'satellite-capsule' && package 'foreman' |                                  |
| External Capsule | !:capsule.internal?       |                                                  |                                  |

Please help me fill the gaps. 
- in what version was 'satellite-capsule' introduced?
- does satellite-capsule imply foreman-proxy and installer scenario 'Capsule'

Reply:
in what version was 'satellite-capsule' introduced? 
-> The package is available since Capsule 6.2 version

does satellite-capsule imply foreman-proxy and installer scenario 'Capsule' 
-> yes its foreman-proxy but only external one. In downstream we only have Capsule distinguishable when its external. This is because on Satellite we by default have foreman-proxy. Yes and the installer scenario Capsule will be for external system with 'satellite-capsule' installed.

Reply:
@upadhyeammit thanks, that means that :capsule.internal? makes no sense. I've updated the table. 

With current deployment model we probably have to care about Sat <= 6.1, correct? Prior 6.2 we had 'capsule-installer' package so I'll update the table again.

Reply:
When we consider having support of Capsule upgrade for 6.7 I think we can skip considering the less than 6.2 versions ? But anyway having condition like below should not be problematic and costly,

package 'satellite-capsule' || package 'capsule-installer'

So I agree to have above condition for Capsule.

Yes internal Capsule does not make sense because we treat that as Satellite.